### PR TITLE
Changes Read to receive uint64 byte count.

### PR DESCRIPTION
**Pull request by ncruces**
Since #2074 changed the type of `Size() uint64` to we should probably change the type of `Read(offset uint32, byteCount uint64) ([]byte, bool)`. Otherwise, there's no API that gets entire buffer, if it covers the entire address range.

As discussed on [slack](https://gophers.slack.com/messages/C04CG4A2NKX/p1708414573291929?thread_ts=1708401526.868449).

### CODE REVIEW

**Comment by mathetake** [comment]
looking good. can you proceed?

**Comment by ncruces** [comment]
I got a bit stuck with the test failures, then work/life got in the way. Let me try again.

**Comment by mathetake** [comment]
yeah cool, no rush anyway!

**Comment by ncruces** [comment]
This will definitely cause more churn than #2074, but if we want to do it, v2 is the right time.

**Review by mathetake — APPROVED** [review]
@evacchi this must be explained in the release note together with https://github.com/tetratelabs/wazero/pull/2074 👍
